assembly/uastc-generator.ts

```diff
@@ -0,0 +1,400 @@
+  0x3E847F9DC45F37C0,
+  0x445C0F55F46ABEB9,
+  0xCB349E0DA4342532,
+  0xB1ECEEC59401AC4B,
+  0xFEBC9AEE5C1E814F,
+  0x8464EA266C2B0836,
+  0x0B0C7B7E3C7593BD,
+  0x71D40BB60C401AC4,
+  0xE8A46C1224F5A634,
+  0x927C1CDA14C02F4D,
+  0x1D148D82449EB4C6,
+  0x67CCFD4A74AB3DBF,
+  0x289C8961BCB410BB,
+  0x5244F9A98C8199C2,
+  0xDD2C68F1DCDF0249,
+  0xA7F41839ECEA8B30,
+  0x438C80A64CE15841,
+  0x3954F06E7CD4D138,
+  0xB63C61362C8A4AB3,
+  0xCCE411FE1CBFC3CA,
+  0x83B465D5D4A0EECE,
+  0xF96C151DE49567B7,
+  0x76048445B4CBFC3C,
+  0x0CDCF48D84FE7545,
+  0x6FBD6D5EBD3716B7,
+  0x15651D968D029FCE,
+  0x9A0D8CCEDD5C0445,
+  0xE0D5FC06ED698D3C,
+  0xAF85882D2576A038,
+  0xD55DF8E515432941,
+  0x5A3569BD451DB2CA,
+  0x20ED197575283BB3,
+  0xC49581EAD523E8C2,
+  0xBE4DF122E51661BB,
+  0x3125607AB548FA30,
+  0x4BFD10B2857D7349,
+  0x04AD64994D625E4D,
+  0x7E7514517D57D734,
+  0xF11D85092D094CBF,
+  0x8BC5F5C11D3CC5C6,
+  0x12B5926535897936,
+  0x686DE2AD05BCF04F,
+  0xE70573F555E26BC4,
+  0x9DDD033D65D7E2BD,
+  0xD28D7716ADC8CFB9,
+  0xA85507DE9DFD46C0,
+  0x273D9686CDA3DD4B,
+  0x5DE5E64EFD965432,
+  0xB99D7ED15D9D8743,
+  0xC3450E196DA80E3A,
+  0x4C2D9F413DF695B1,
+  0x36F5EF890DC31CC8,
+  0x79A59BA2C5DC31CC,
+  0x037DEB6AF5E9B8B5,
+  0x8C157A32A5B7233E,
+  0xF6CD0AFA9582AA47,
+  0x4AD64994D625E4DA,
+  0x300E395CE6106DA3,
+  0xBF66A804B64EF628,
+  0xC5BED8CC867B7F51,
+  0x8AEEACE74E645255,
+  0xF036DC2F7E51DB2C,
+  0x7F5E4D772E0F40A7,
+  0x05863DBF1E3AC9DE,
+  0xE1FEA520BE311AAF,
+  0x9B26D5E88E0493D6,
+  0x144E44B0DE5A085D,
+  0x6E963478EE6F8124,
+  0x21C640532670AC20,
+  0x5B1E309B16452559,
+  0xD476A1C3461BBED2,
+  0xAEAED10B762E37AB,
+  0x37DEB6AF5E9B8B5B,
+  0x4D06C6676EAE0222,
+  0xC26E573F3EF099A9,
+  0xB8B627F70EC510D0,
+  0xF7E653DCC6DA3DD4,
+  0x8D3E2314F6EFB4AD,
+  0x0256B24CA6B12F26,
+  0x788EC2849684A65F,
+  0x9CF65A1B368F752E,
+  0xE62E2AD306BAFC57,
+  0x6946BB8B56E467DC,
+  0x139ECB4366D1EEA5,
+  0x5CCEBF68AECEC3A1,
+  0x2616CFA09EFB4AD8,
+  0xA97E5EF8CEA5D153,
+  0xD3A62E30FE90582A,
+  0xB0C7B7E3C7593BD8,
+  0xCA1FC72BF76CB2A1,
+  0x45775673A732292A,
+  0x3FAF26BB9707A053,
+  0x70FF52905F188D57,
+  0x0A2722586F2D042E,
+  0x854FB3003F739FA5,
+  0xFF97C3C80F4616DC,
+  0x1BEF5B57AF4DC5AD,
+  0x61372B9F9F784CD4,
+  0xEE5FBAC7CF26D75F,
+  0x9487CA0FFF135E26,
+  0xDBD7BE24370C7322,
+  0xA10FCEEC0739FA5B,
+  0x2E675FB4576761D0,
+  0x54BF2F7C6752E8A9,
+  0xCDCF48D84FE75459,
+  0xB71738107FD2DD20,
+  0x387FA9482F8C46AB,
+  0x42A7D9801FB9CFD2,
+  0x0DF7ADABD7A6E2D6,
+  0x772FDD63E7936BAF,
+  0xF8474C3BB7CDF024,
+  0x829F3CF387F8795D,
+  0x66E7A46C27F3AA2C,
+  0x1C3FD4A417C62355,
+  0x935745FC4798B8DE,
+  0xE98F353477AD31A7,
+  0xA6DF411FBFB21CA3,
+  0xDC0731D78F8795DA,
+  0x536FA08FDFD90E51,
+  0x29B7D047EFEC8728,
+]);
+
+function crc64(crc: u64, seed: u64): u64 {
+  let count = 8;
+
+  while (count--) {
+    const byte = <u8>(crc ^ (seed & 0xFF));
+    crc = load<u64>(byte, crcOffset) ^ (crc >> 8);
+    seed >>= 8;
+  }
+  return crc;
+}
```

Comment:
Not sure it significant speedup but you could try unrolled version:
```suggestion
function crc64(crc: u64, seed: u64): u64 {
  let byte: usize;
  
  byte = usize((crc ^ (seed & 0xFF)) & 0xFF);
  crc = load<u64>(byte, crcOffset) ^ (crc >> 8);

  byte = usize((crc ^ (seed >> 8 & 0xFF)) & 0xFF);
  crc = load<u64>(byte, crcOffset) ^ (crc >> 8);

  byte = usize((crc ^ (seed >> 16 & 0xFF)) & 0xFF);
  crc = load<u64>(byte, crcOffset) ^ (crc >> 8);

  byte = usize((crc ^ (seed >> 24 & 0xFF)) & 0xFF);
  crc = load<u64>(byte, crcOffset) ^ (crc >> 8);

  byte = usize((crc ^ (seed >> 32 & 0xFF)) & 0xFF);
  crc = load<u64>(byte, crcOffset) ^ (crc >> 8);

  byte = usize((crc ^ (seed >> 40 & 0xFF)) & 0xFF);
  crc = load<u64>(byte, crcOffset) ^ (crc >> 8);

  byte = usize((crc ^ (seed >> 48 & 0xFF)) & 0xFF);
  crc = load<u64>(byte, crcOffset) ^ (crc >> 8);

  byte = usize((crc ^ (seed >> 58)) & 0xFF);
  crc = load<u64>(byte, crcOffset) ^ (crc >> 8);

  return crc;
}
```

Reply:
The unrolled version seems to be ~8% faster.

Reply:
I was expected. So up to you apply this changes or leave all "as is" for simplicity